subprojects/gradle/namespaced-resources-fixer/src/test/kotlin/com/avito/android/plugin/FileResourcesFixerTest.kt

```diff
@@ -0,0 +1,147 @@
+package com.avito.android.plugin
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+class FileResourcesFixerTest {
+
+    private lateinit var file: File
+
+    @BeforeEach
+    fun setup(@TempDir tempDir: File) {
+        file = File(tempDir, "File.kt")
+    }
+
+    @Test
+    fun `change nothing - has no import with merged R class`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = emptyList<AndroidModule>()
+        val result = fix(
+            """
+            class Screen {}
+        """,
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            class Screen {}
+        """
+        )
+    }
+
+    @Test
+    fun `change nothing - not used import`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = emptyList<AndroidModule>()
+        val result = fix(
+            """
+            import com.avito.android.R
+            class Screen {}
+        """,
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
+            class Screen {}
+        """
+        )
+    }
+
+    @Test
+    fun `change nothing - ambiguous imports`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = listOf(
+            AndroidModule("com.avito.android.a", listOf("container")),
+            AndroidModule("com.avito.android.b", listOf("container"))
+        )
+        val result = fix(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """,
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """
+        )
+    }
+
+    @Test
+    fun `replace library id - simple property`() {
+        val app = AndroidModule("com.avito.android.R", emptyList())
+        val libraries = listOf(
+            AndroidModule("com.avito.android.library", listOf("container"))
+        )
+        val result = fix(
+            """
+            import com.avito.android.R
+
+            class Screen {
+                val id = R.id.container
+            }
+        """.trimIndent(),
+            app, libraries
+        )
+        assertThat(result).isEqualTo(
+            """
+            import com.avito.android.R
```

Comment:
Why do we need that import in the result?

Reply:
I want to postpone this optimization with a merged R class.

It's a rare case when we have only unambiguous usages and can remove merged R class after such optimization.

As a next improvement, I want to rename it:

```kotlin
import com.avito.android.R as deprecated_merged_R
```